FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureTableClientFactory.cs

```diff
@@ -16,19 +16,19 @@ public class AzureTableClientFactory : IAzureTableClientFactory
     {
         private const string StorageEmulatorConnectionString = "UseDevelopmentStorage=true";
         private readonly ITokenCredentialProvider _credentialProvider;
-
         private readonly string _tableUrl;
+        private readonly string _internalConnectionString;
 
         public AzureTableClientFactory(
-            IOptions<JobConfiguration> config,
+            IOptions<StorageConfiguration> storageConfiguration,
+            IOptions<JobConfiguration> jobConfiguration,
             ITokenCredentialProvider credentialProvider)
         {
-            EnsureArg.IsNotNull(config, nameof(config));
-            EnsureArg.IsNotNullOrWhiteSpace(config.Value.TableUrl, nameof(config.Value.TableUrl));
-
-            _tableUrl = config.Value.TableUrl;
-
+            EnsureArg.IsNotNull(storageConfiguration, nameof(storageConfiguration));
+            _tableUrl = EnsureArg.IsNotNullOrWhiteSpace(jobConfiguration?.Value?.TableUrl, nameof(jobConfiguration.Value.TableUrl));
```

Comment:
need to make sure that both `_tableUrl` and `_internalConnectionString` should not be null at the same time

Reply:
Updated here, thanks.